node/cn/tracers/api.go

```diff
@@ -266,8 +334,9 @@ func (api *PrivateDebugAPI) traceChain(ctx context.Context, start, end *types.Bl
 			}
 			close(results)
 		}()
+		var preferDisk bool
 		// Feed all the blocks both into the tracer, as well as fast process concurrently
-		for number = start.NumberU64() + 1; number <= end.NumberU64(); number++ {
+		for number = start.NumberU64(); number < end.NumberU64(); number++ {
```

Comment:
Is this boundary okay? It seems previous one includes `end`, but not here.

Reply:
Yes, the boundary is not changed. The target tracing block is started from `start+1`. You can see the target block is retrieved by calling `rpc.BlockNumber(number+1)`
```
	next, err := api.blockByNumber(localctx, rpc.BlockNumber(number+1))
	if err != nil {
		failed = err
		break
	}
	// Send the block over to the concurrent tracers (if not in the fast-forward phase)
	txs := next.Transactions()
	if notifier != nil {
		select {
		case tasks <- &blockTraceTask{statedb: statedb.Copy(), block: next, rootref: block.Root(), results: make([]*txTraceResult, len(txs))}:
		case <-notifier.Closed():
			return
		}
	} else {
		tasks <- &blockTraceTask{statedb: statedb.Copy(), block: next, rootref: block.Root(), results: make([]*txTraceResult, len(txs))}
	}
```

Reply:
@jeongkyun-oh Ignore the explanation in the previous comment. I looked again and figured out that tracing is actually performed from when number is greater than origin. Because of this, the actual tracing range is [start+1,end].